examples/amm/src/contract.rs

```diff
@@ -220,87 +262,348 @@ impl AmmContract {
 
                 let other_amount = if token_to_remove_idx == 0 {
                     Amount::from_attos(
-                        ((token_to_remove_amount_bigint * balance1_bigint) / balance0_bigint)
-                            .to_u128()
-                            .expect("Couldn't convert other_amount to u128"),
+                        ((token_to_remove_amount_bigint * balance1_bigint.clone())
+                            / balance0_bigint.clone())
```

Comment:
```suggestion
                        ((token_to_remove_amount_bigint * &balance1_bigint)
                            / &balance0_bigint)
```
(Also below. I don't think it's ever necessary to clone a bigint to use it.)

Reply:
I didn't think so either, but apparently everytime you use them they get moved 😅